### PR TITLE
Fix clippy

### PR DESCRIPTION
**Pull request by darnuria**
Some commits to fix CI breakage of clippy, feel free to pick what's pertinant or roll'up. I didn't make the derive `Eq` suggestion because didn't know what is the policy about `Eq`.

### CODE REVIEW

**Comment by darnuria** [comment]
Also had some issue on a suggestion I have a "less verbose" change who keep clippy happy, but at first clippy when crazy with his suggestion. see https://github.com/rust-lang/rust-clippy/issues/8281#issuecomment-1212597668.

My suggestion would be to do:

```diff
diff --git a/crates/client/src/client/async_client.rs b/crates/client/src/client/async_client.rs
index 103bc102..7f49a896 100644
--- a/crates/client/src/client/async_client.rs
+++ b/crates/client/src/client/async_client.rs
@@ -852,13 +852,7 @@ where
 
         let message = if let Some(response) = ready!(self.state.inner().poll_next_unpin(cx)) {
             Some(match response {
-                Ok(ok) => {
-                    if let Err(e) = self.state.process(ok.answers()) {
-                        Err(e)
-                    } else {
-                        Ok(ok)
-                    }
-                }
+                Ok(ok) => self.state.process(ok.answers()).map(|_| ok),
                 Err(e) => Err(e.into()),
             })
         } else {
```

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Your suggestion looks good to me, thanks for cleaning this up!

**Comment by darnuria** [comment]
> Your suggestion looks good to me, thanks for cleaning this up!
 Will do a commit about it, and for the `derive` `Eq` if `PartialEq` clippy thing I have no real idea of the implication tell me if it's OK to make it ! :)

File: crates/client/src/client/async_client.rs [hunk 1]
```diff
@@ -851,20 +850,11 @@ where
...
-                Err(e) => Err(e.into()),
-            })
-        } else {
-            None
-        };
+        let message = ready!(self.state.inner().poll_next_unpin(cx)).map(|response| {
+            let ok = response?;
+            self.state.process(ok.answers())?;
+            Ok(ok)
+        });
```
Comment (darnuria) [inline crates/client/src/client/async_client.rs:857]: @djc finally found a compact and clear way to do it. When `try { }` will land the awkward` `map` would retire. ;)

May need a carefull review.

Comment (darnuria) [reply crates/client/src/client/async_client.rs:857]: if needed `ready!` result could be in a separated `let` i am quite a fan of binding everything it help a lot debugging experience.

Comment (bluejekyll) [reply crates/client/src/client/async_client.rs:857]: This looks like a good cleanup.

**Review by bluejekyll — APPROVED** [review]
This looks really good. Thanks for the PR.

**Comment by bluejekyll** [comment]
Merging. Thanks for the PR!

**Comment by darnuria** [comment]
Thanks you both for review time!

**Pull request merged**